sdk/go/common/resource/config/plaintext.go

```diff
@@ -0,0 +1,187 @@
+func NewSecurePlaintext(plaintext string) Plaintext {
+	return Plaintext{value: plaintext, secure: true}
+}
+
+// Secure returns true if the receiver is a secure string or a composite value that contains a secure string.
+func (c Plaintext) Secure() bool {
+	switch v := c.Value().(type) {
+	case []Plaintext:
+		for _, v := range v {
+			if v.Secure() {
+				return true
+			}
+		}
+		return false
+	case map[string]Plaintext:
+		for _, v := range v {
+			if v.Secure() {
+				return true
+			}
+		}
+		return false
+	case string:
+		return c.secure
+	default:
+		return false
+	}
+}
+
+// Value returns the inner plaintext value.
+func (c Plaintext) Value() any {
+	return c.value
+}
+
+// GoValue returns the inner plaintext value as a plain Go value:
+//
+// - secure strings are mapped to their plaintext
+// - []Plaintext values are mapped to []any values
+// - map[string]Plaintext values are mapped to map[string]any values
+func (c Plaintext) GoValue() any {
+	switch v := c.Value().(type) {
+	case []Plaintext:
+		vs := make([]any, len(v))
+		for i, v := range v {
+			vs[i] = v.GoValue()
+		}
+		return vs
+	case map[string]Plaintext:
+		vs := make(map[string]any, len(v))
+		for k, v := range v {
+			vs[k] = v.GoValue()
+		}
+		return vs
+	default:
+		return v
+	}
+}
+
+// Encrypt converts the receiver as a Value. All secure strings in the result are encrypted using encrypter.
+func (c Plaintext) Encrypt(ctx context.Context, encrypter Encrypter) (Value, error) {
+	obj, err := c.encrypt(ctx, encrypter)
+	if err != nil {
+		return Value{}, err
+	}
+	return obj.marshalValue()
+}
+
+// encrypt converts the receiver to an object. All secure strings in the result are encrypted using encrypter.
+func (c Plaintext) encrypt(ctx context.Context, encrypter Encrypter) (object, error) {
+	switch v := c.Value().(type) {
+	case bool:
+		return newObject(v), nil
+	case int64:
+		return newObject(v), nil
+	case float64:
+		return newObject(v), nil
+	case string:
+		if !c.secure {
+			return newObject(v), nil
+		}
+		ciphertext, err := encrypter.EncryptValue(ctx, v)
+		if err != nil {
+			return object{}, err
+		}
+		return newSecureObject(ciphertext), nil
+	case []Plaintext:
+		vs := make([]object, len(v))
+		for i, v := range v {
+			ev, err := v.encrypt(ctx, encrypter)
+			if err != nil {
+				return object{}, err
+			}
+			vs[i] = ev
+		}
+		return newObject(vs), nil
+	case map[string]Plaintext:
+		vs := make(map[string]object, len(v))
+		for k, v := range v {
+			ev, err := v.encrypt(ctx, encrypter)
+			if err != nil {
+				return object{}, err
+			}
+			vs[k] = ev
+		}
+		return newObject(vs), nil
+	default:
+		contract.Failf("unexpected plaintext of type %T", v)
+		return object{}, nil
+	}
+}
+
+// marshalText returns the text representation of the plaintext.
+func (c Plaintext) marshalText() (string, error) {
+	if str, ok := c.Value().(string); ok {
+		return str, nil
+	}
+	bytes, err := json.Marshal(c.GoValue())
```

Comment:
Won't this panic? `GoValue()` will return a `Plaintext`, and `json.Marshal` will call `MarshalJson`.

Reply:
> Won't this panic? GoValue() will return a Plaintext,

`GoValue` does not return a `Plaintext`. It returns a JSON-compatible value.

Reply:
You're right. Ignore this comment. 😅